RPGGame/GameObject/Entity.cs

```diff
@@ -13,5 +13,26 @@ public class Entity(Vector2 position, Vector2 size)
 
         [JsonProperty]
         public Vector2 Size { get; protected set; } = size;
+
+        public virtual bool Move(Vector2 targetPos, bool relative)
+        {
+            bool success = true;
+
+            if (relative)
+            {
+                targetPos += Position;
+            }
+
+            if (targetPos.X > 0 && targetPos.Y > 0)
+            {
+                Position = targetPos;
+            }
+            else
+            {
+                success = false;
+            }
+           
+            return success;
```

Comment:
IMO using an early return is a cleaner way of doing this, especially accounting for future expansion. Up to you though.
```suggestion
            if (relative)
            {
                targetPos += Position;
            }

            if (targetPos.X < 0 || targetPos.Y < 0)
            {
                return false;
            }

            Position = targetPos;
            return true;
```

Reply:
i didn't even realise that I'd used an AND instead of an OR operator... oops. but I'm down to add whatever you think is best, my initial logic behind using the variable was that I thought it might've made it easier to add different conditions that set the success variable to false in the future, but I realise now my logic was floored anyway. infact, im not sure where my head was the whole time i was writing this